src/lib/middleware/featureFlagMiddleware.ts

```diff
@@ -0,0 +1,48 @@
+import { NextFunction } from "express"
+import { ArtsyRequest, ArtsyResponse } from "./artsyExpress"
+import {
+  createFeatureFlagService,
+  FeatureFlagService,
+} from "lib/featureFlags/featureFlagService"
+
+export function featureFlagMiddleware(serviceType: symbol) {
+  let service
+  return (req: ArtsyRequest, res: ArtsyResponse, next: NextFunction) => {
+    new Promise<FeatureFlagService>(async (resolve, reject) => {
+      if (service) {
+        resolve(service)
+      } else {
+        try {
+          service = await createFeatureFlagService(serviceType)
+          resolve(service)
+        } catch {
+          reject("An unknown error occurred while creating the flag service.")
```

Comment:
Why not reject the actual error (which is known, because if it were not then the `catch` block wouldn't exectute)? 

```
try {
  ...
} catch (error) {
  reject(error)
}
```
This is then handled below at line 44 (see comment)